torchtext/transforms.py

```diff
@@ -189,3 +193,350 @@ def forward(self, input: Any) -> Any:
+            bpe token ids. For example: {"ingly": 4420}
+    :type bpe_encoder: Dict[str, int]
+    :param bpe_merge_ranks: A dict mapping a pair of bpe tokens to rank.
+            Torchscript doesn't support tuple as dict key type, concat
+            a pair of string with SEPERATOR (e.g \u0001) as workaround.
+            For example: {"ing\u0001ly": 210}
+    :type bpe_merge_ranks: Dict[str, int]
+    :param byte_encoder: A dict mapping a byte to an unicode character.
+    :type byte_encoder: Dict[int, str]
+    """
+    SEPERATOR: str = "\u0001"
+    LSTRIP_PATTERN: str = "\u0120"
+
+    def __init__(
+        self,
+        bpe_encoder: Dict[str, int],
+        bpe_merge_ranks: Dict[str, int],
+        byte_encoder: Dict[int, str],
+    ):
+        super().__init__()
+
+        self.bpe_encoder = bpe_encoder
+        self.bpe_merge_ranks = bpe_merge_ranks
+        self.byte_encoder = byte_encoder
+
+        self.bpe_decoder = self.load_bpe_decoder(bpe_encoder)
+        self.byte_decoder = self.load_byte_decoder(byte_encoder)
+
+        self.merge_seperator = self.SEPERATOR
+        self.lstrip_pattern = self.LSTRIP_PATTERN
+        self.inf = len(self.bpe_merge_ranks) + 1
+
+    @classmethod
+    def get_gpt2_bpe_tokenizer(
+        self,
+        encoder_json_path: str,
+        vocab_bpe_path: str,
+    ):
+        if os.path.exists(encoder_json_path):
+            encoder_json_local_path = encoder_json_path
+        else:
+            encoder_json_local_path = download_from_url(
+                url=encoder_json_path,
+                root=_CACHE_DIR
+            )
+        bpe_encoder = GPT2BPETokenizer.load_bpe_encoder(encoder_json_local_path)
+
+        if os.path.exists(vocab_bpe_path):
+            vocab_bpe_local_path = vocab_bpe_path
+        else:
+            vocab_bpe_local_path = download_from_url(
+                url=vocab_bpe_path,
+                root=_CACHE_DIR
+            )
+        bpe_merge_ranks = GPT2BPETokenizer.load_bpe_vocab(vocab_bpe_local_path)
+        return GPT2BPETokenizer(
+            bpe_encoder,
+            bpe_merge_ranks,
+            GPT2BPETokenizer.load_byte_encoder()
+        )
+
+    @classmethod
+    def load_bpe_encoder(cls, bpe_encoder_path: str) -> Dict[str, int]:
+        if os.path.exists(bpe_encoder_path):
+            local_path = bpe_encoder_path
+        else:
+            local_path = download_from_url(url=bpe_encoder_path, root=_CACHE_DIR)
+        with open(local_path, "r", encoding="utf-8") as f:
+            return json.load(f)
+
+    @classmethod
+    def load_bpe_vocab(cls, bpe_vocab_path: str) -> Dict[str, int]:
+        if os.path.exists(bpe_vocab_path):
+            local_path = bpe_vocab_path
+        else:
+            local_path = download_from_url(url=bpe_vocab_path, root=_CACHE_DIR)
+        with open(local_path, "r", encoding="utf-8") as f:
+            bpe_vocab = f.read()
+
+        return {
+            cls.SEPERATOR.join(merge_pair.split()): i
```

Comment:
Curious, why BPE file cannot be formatted as JSON? 
If bpe_vocab and bpe_encoder are in same format, then they can share the same I/O logic.

Reply:
Technically, nothings stops us from formatting them as a JSON. However, if I think from a user's perspective, we should align with the file formats that are released by AllenAI, HuggingFace, etc. If people train their own GPT-2 tokenizers using the original implementation of any of these libraries, it is likely that the files produced by them will be in the same format. Hence, I suggest keeping the same file format and letting the implementation deal with Torchscript-specific nuances. Hope that makes sense?